tests/pipelines/test_pipeline_utils.py

```diff
@@ -0,0 +1,123 @@
+import unittest
+
+from diffusers.pipelines.pipeline_utils import is_safetensors_compatible
+
+
+class IsSafetensorsCompatibleTests(unittest.TestCase):
+    def test_is_compatible_all_have_safetensors(self):
+        filenames = [
+            "safety_checker/pytorch_model.bin",
+            "safety_checker/model.safetensors",
+            "vae/diffusion_pytorch_model.bin",
+            "vae/diffusion_pytorch_model.safetensors",
+            "text_encoder/pytorch_model.bin",
+            "text_encoder/model.safetensors",
+            "unet/diffusion_pytorch_model.bin",
+            "unet/diffusion_pytorch_model.safetensors",
+        ]
+        self.assertTrue(is_safetensors_compatible(filenames))
+
+    def test_is_compatible_only_one_safetensors(self):
+        filenames = [
+            "safety_checker/pytorch_model.bin",
+            # Removed: "safety_checker/model.safetensors",
+            "vae/diffusion_pytorch_model.bin",
+            # Removed: "vae/diffusion_pytorch_model.safetensors",
+            "text_encoder/pytorch_model.bin",
+            # Removed: "text_encoder/model.safetensors",
+            "unet/diffusion_pytorch_model.bin",
+            "unet/diffusion_pytorch_model.safetensors",  # <- Only matching safetensors file
+        ]
+        self.assertTrue(is_safetensors_compatible(filenames))
```

Comment:
I was under the impression this shouldn't be true. But I could be wrong.

Reply:
Yes same, I think for now this should not be allowed.

Reply:
Ah sorry, I mixed up the variant logic with the safetensors logic. We use a variant if there's at least one matching variant file but we only use safetensors if there's a matching safetensor's file for each file. My mistake, will fix!

Reply:
fixed!